includes/downloader.js

```diff
@@ -113,3 +112,58 @@ function getTimestampString() {
 }
 
 
+function downloadCSV() {
+    const columnNames = INA.columnNames;  
+    const statements = INA.statements;    
+
+    // Validation: Check if columnNames or statements are empty
+    if (!columnNames || columnNames.length === 0 || !statements || statements.length === 0) {
+        alert("No data found.");
+        return;  // Exit the function if no data
+    }
+
+    // Start CSV content with column names (header row)
+    let csvContent = columnNames.join(",") + "\n";
+
+    // Loop through each statement object
+    statements.forEach(statement => {
+        // For each statement, create a row by joining values in the same order as the column names
+        let row = columnNames.map(col => {
+            // Handle undefined or null values, and escape commas if needed
+            const value = statement[col] ? statement[col] : '';
+            return `"${value.toString().replace(/"/g, '""')}"`;  // Escape quotes
```

Comment:
> * I'm not sure where this happens, but when downloading the CSV I get extra lines that only contain a double quote `"`:

I think what is happening is that instead of adding an empty quote, the trailing newline is getting wrapped in quotes `\n` -> `"\n"`.

I'm not exactly sure yet how this is happening (don't know the behaviour of `replace` in JS), but I think doing the following would do it
```javascript
if (value == '') {
    return value;
} else {
    return ...; // escape
}
```

Reply:
Actually I'm less sure now, maybe my analysis is incorrect :frowning_face: